crates/turborepo-lib/src/engine/builder.rs

```diff
@@ -37,7 +37,7 @@ pub struct EngineBuilder<'a> {
     package_graph: &'a PackageGraph,
     is_single: bool,
     turbo_jsons: Option<HashMap<WorkspaceName, TurboJson>>,
-    workspaces: Vec<WorkspaceName>,
+    workspaces: Vec<&'a WorkspaceName>,
```

Comment:
The apostrophe is what Rust uses to distinguish types from lifetimes in generics e.g. `a` would be a type, but `'a` is a lifetime. The change from `Vec<WorkspaceName>` to `Vec<&'a WorkspaceName>` is a change from a vector of ["owned"](https://doc.rust-lang.org/rust-by-example/scope/move.html) workspace names to ["borrowed"](https://doc.rust-lang.org/rust-by-example/scope/borrow.html) workspace names. Specifically, the borrow is valid for the length of lifetime `'a`.

Reply:
This is to avoid copies. At a certain point in a run, all of the data exists and is stored in structs, so it makes sense to reference it instead of copying it into other structs. In the future we'll have a little more of a consistent story for lifetimes, i.e. we'll have a single lifetime for a run that we can use to consistently store references.